src/dbnode/persist/fs/write.go

```diff
@@ -390,14 +399,23 @@ func (w *writer) closeWOIndex() error {
 		return err
 	}
 
-	return closeAll(
+	fds := []digest.FdWithDigestWriter{
 		w.infoFdWithDigest,
 		w.indexFdWithDigest,
 		w.summariesFdWithDigest,
 		w.bloomFilterFdWithDigest,
 		w.dataFdWithDigest,
 		w.digestFdWithDigestContents,
-	)
+	}
+
+	if w.syncBeforeClose {
+		if err := digest.SyncAll(fds...); err != nil {
+			_ = digest.CloseAll(fds...)
```

Comment:
nit; probably don't want to lose the error here? May be worth logging it at least, if not adding it to a full multierr

Reply:
Will add it to the `MultiError`.